python/ql/test/library-tests/frameworks/django-v2-v3/routing_test.py

```diff
@@ -150,3 +150,40 @@ def get(self, request): # $ requestHandler
 urlpatterns = [
     path("UnknownViewSubclass/", UnknownViewSubclass.as_view()),  # $ routeSetup="UnknownViewSubclass/"
 ]
+
+################################################################################
+# Routing to *args and **kwargs
+################################################################################
+
+def kwargs_param(request, **kwargs): # $ requestHandler routedParameter=kwargs
+    ensure_tainted(
+        kwargs, # $ tainted
+        kwargs["foo"], # $ tainted
+        kwargs["bar"]  # $ tainted
+    )
+
+    ensure_tainted(request) # $ tainted
+
+
+def star_args_param(request, *args): # $ requestHandler routedParameter=args
+    ensure_tainted(
+        args, # $ tainted
+        args[0], # $ tainted
+        args[1], # $ tainted
+    )
+    ensure_tainted(request) # $ tainted
+
+
+def star_args_param_check(request, foo, bar): # $ requestHandler routedParameter=foo routedParameter=bar
+    ensure_tainted(
+        foo, # $ tainted
+        bar, # $ tainted
+    )
+    ensure_tainted(request) # $ tainted
```

Comment:
I am not sure what this is testing? That `foo` and `bar` are routed? Is that not established by earlier tests?

Reply:
I guess you can call this double testing :shrug: for the uninitiated it might not be given that all routed parameters are also remote-flow-sources, so I don't see any harm in having the test written like this?

Reply:
Fair enough 🙂